packages/hdwallet-keepkey/src/ethereum.ts

```diff
@@ -21,8 +21,8 @@ export function ethSupportsNativeShapeShift(): boolean {
   return true;
 }
 
-export function ethSupportsEIP1559(): boolean {
-  return this.ethSupportsEIP1559();
+export async function ethSupportsEIP1559(): Promise<boolean> {
+  return await this.ethSupportsEIP1559();
```

Comment:
This breaks the way all of the "Supports" functions work. This is why we did firmware checks in axiom instead of hdwallet.

I don't like the idea of having some "Supports" functions be async and some be sync.

Reply:
"Supports" functions are all hard coded booleans which doesn't seem right either. From my point of view, directly querying the wallet is the most direct, efficient and understandable way to obtain feature support, especially for the hardware wallets.

Reply:
the drawback of checking in Axiom is all other clients would need to do the same

Reply:
> I don't like the idea of having some "Supports" functions be async and some be sync.

agreed.  can we make them all async?